From 9d939fb20675f0f729dc4f7165d896e1d53a4603 Mon Sep 17 00:00:00 2001
From: RadioNoiseE <rne.kou@icloud.com>
Date: Tue, 11 Nov 2025 18:46:26 +0800
Subject: [PATCH] Fix indentation for CDATA section content

* lisp/nxml/nxml-mode.el (nxml-merge-indent-context-type): Handle
'cdata-section type to prevent it from returning 'mixed.
* lisp/nxml/nxml-mode.el (nxml-compute-indent-in-delimited-token):
Preserve existing indentation for CDATA section content.
---
 lisp/nxml/nxml-mode.el | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/lisp/nxml/nxml-mode.el b/lisp/nxml/nxml-mode.el
index 9b780a9..b166615 100644
--- a/lisp/nxml/nxml-mode.el
+++ b/lisp/nxml/nxml-mode.el
@@ -1430,6 +1430,8 @@ either nil or one of the symbols `start-tag', `end-tag', `markup',
 	 (if (memq context '(nil end-tag comment))
 	     'end-tag
 	   'mixed))
+        ((eq xmltok-type 'cdata-section)
+         (or context 'markup))
 	((eq xmltok-type 'comment)
 	 (cond ((memq context '(start-tag end-tag comment))
 		context)
@@ -1519,7 +1521,9 @@ OPEN-DELIM and CLOSE-DELIM are strings giving the opening and closing
 delimiters.  POS is the position of the first non-whitespace character
 of the line.  This expects the xmltok-* variables to be set up as by
 `xmltok-forward'."
-  (cond ((let ((end (+ pos (length close-delim))))
+  (cond ((string= open-delim "<![CDATA[")
+         (goto-char pos))
+        ((let ((end (+ pos (length close-delim))))
 	   (and (<= end (point-max))
 		(string= (buffer-substring-no-properties pos end)
 			 close-delim)))
-- 
2.50.1 (Apple Git-155)

